react-app/src/components/projects/EnhancedReferralDates.tsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import { Box } from '@mui/material';
+import { LocalizationProvider } from '@mui/x-date-pickers';
+import { DateField } from '@mui/x-date-pickers/DateField';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import dayjs from 'dayjs';
+import { NotificationQueue } from '@/hooks/api/useProjectNotificationApi';
+import { NotificationType } from '@/constants/notificationTypes';
+
+interface EnhancedReferralDatesProps {
+  rows: NotificationQueue[];
+}
+
+const EnhancedReferralDates: React.FC<EnhancedReferralDatesProps> = ({ rows }) => {
```

Comment:
We could add a default here to prevent any errors if the passed rows is undefined.
```suggestion
const EnhancedReferralDates: React.FC<EnhancedReferralDatesProps> = ({ rows = [] }) => {
```